### PR TITLE
feat(post): add 16k page size

### PR DESCRIPTION
**Pull request by RossComputerGuy**

### CODE REVIEW

File: content/post/why-16k-page-size/index.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+description: 16k vs 4k page sizes & Asahi Linux.
+date: 2024-06-09 18:27:00 +0800
+---
+
+Since August of 2023, I've been a daily user of a 14-inch 2021 Apple MacBook Pro (M1 Pro, 16GB RAM) laptop.
+Thanks to [@tpwrules](https://github.com/tpwrules)'s [NixOS Asahi Linux Flake](https://github.com/tpwrules/nixos-apple-silicon/tree/main),
+I was able to set up the laptop to run NixOS. The majority of software have worked perfectly on it and I
+consider it to be the best Linux laptop for its performance. However, one major issue isn't a hardware issue
+but a software one. When you see memory sizes on a system, you commonly think about things like the timing,
+what kind of RAM chips they use, the number of RAM chips they use, and so on. Though, this isn't that.
```
Comment (Enstrayed) [inline content/post/why-16k-page-size/index.md]: `number of RAM chips they use` have?

File: content/post/why-16k-page-size/index.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+was first introduced along with Windows 95. However, our memory sizes nowadays are vastly larger.
+This means for a program to allocate something like 1GB of memory, they will need to ask the operating system
+for a certain number of pages which gets them up to 1GB. On a 4k page size system, this would be about 250 pages
+of memory. However, on a 16k page size system, this would be 63. Despite the math being 62.5, only whole pages
+can be allocated. This means the memory allocator used will have to keep that last page for other memory allocations.
+Between the 4k and 16k page sizes, it is clear that allocating a smaller number of pages can be beneficial when you
+utilize programs that often are going to be allocating memory sizes beyond 1MB. This can result in a performance
+improvement as there will be less work the CPU will have to do to ensure the program gets the correct number of pages.
+
+Due to Asahi Linux using 16k page size memory, this means programs not coded to expect a 14k page size will fail.
```
Comment (Enstrayed) [inline content/post/why-16k-page-size/index.md]: `14k` ?

File: content/post/why-16k-page-size/index.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+what kind of RAM chips they use, the number of RAM chips they use, and so on. Though, this isn't that.
+This particular issue is one designed into the Asahi Linux kernel and even macOS's Darwin kernel.
+This would be the page size, the number which dictates the minimum size of a page of memory is. Every
+modern computer nowadays, since the 90's, use something called page memory. This is a technique to
+efficiently manage memory but it is also used as a security mechanism. Programs can be given permissions
+to only certain regions of memory via page memory. However, this exact issue is not an issue with the security
+but the size.
+
+The vast majority of users use x86 which utilizes 4k page memory. This size was great when page memory
+was first introduced along with Windows 95. However, our memory sizes nowadays are vastly larger.
```
Comment (Enstrayed) [inline content/post/why-16k-page-size/index.md]: `was first introduced along with Windows 95` introduced in the Win95 era or explicitly *with* Win95? I assume the former

File: content/post/why-16k-page-size/index.md (hunk 1 above)
Comment (RossComputerGuy) [reply content/post/why-16k-page-size/index.md]: Yes, the amount of RAM chips in a stick of memory or in each channel matters.

File: content/post/why-16k-page-size/index.md (hunk 2 above)
Comment (RossComputerGuy) [reply content/post/why-16k-page-size/index.md]: Typo.

File: content/post/why-16k-page-size/index.md (hunk 3 above)
Comment (RossComputerGuy) [reply content/post/why-16k-page-size/index.md]: Win95 afaik is the first major consumer version of Windows to utilize protected mode and user mode which involves paging. 32-bit x86 CPU's specifically.

File: content/post/why-16k-page-size/index.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+what kind of RAM chips they use, the number of RAM chips in a stick or channel of memory, and so on.
+Though, this isn't that. This particular issue is one designed into the Asahi Linux kernel and even macOS's
+Darwin kernel. This would be the page size, the number which dictates the minimum size of a page of memory is.
+Every modern computer nowadays, since the 90's, use something called page memory. This is a technique to
+efficiently manage memory but it is also used as a security mechanism. Programs can be given permissions
+to only certain regions of memory via page memory. However, this exact issue is not an issue with the security
+but the size.
+
+The vast majority of users use x86 which utilizes 4k page memory. This size was great when page memory
+was first introduced along with hardware to run Windows 95. However, our memory sizes nowadays are vastly larger.
```
Comment (Enstrayed) [inline content/post/why-16k-page-size/index.md]: Trying to lookup info about when paging was introduced is annoying as the semantics about what "paging" means seems to have confused the Wikipedia editors. It *might* have been introduced with the 386 but that might be something entirely different. As such, ¯\_(ツ)_/¯

**Review by Enstrayed — APPROVED** [review]

File: content/post/why-16k-page-size/index.md (hunk 4 above)
Comment (RossComputerGuy) [reply content/post/why-16k-page-size/index.md]: Yes, the i386 is the first in x86 to come with the MMU and paging system.

Comment (RossComputerGuy) [reply content/post/why-16k-page-size/index.md]: Alright, I clarified it a bit and corrected it.

**Pull request merged**